src/mrsimulator/method/__init__.py

```diff
@@ -383,6 +478,76 @@ def get_transition_pathways(self, spin_system) -> List[TransitionPathway]:
             for item in segments
         ]
 
+    def events_to_dataframe(self, properties=None) -> pd.DataFrame:
+        """Returns dataframe of requested Event properties
+
+        Args:
+            List properties: properties to include in df columns. Include all if empty
+
+        Returns:
+            pd.DataFrame df: properties as columns and event number as row
+
+        Example:
+            TODO add example code
+        """
+        # TODO: Add catch for empty 'spectral_dimensions' and 'events'
+        spec_dims = self.spectral_dimensions
+        gsp = self.get_symmetry_pathways
+
+        df = pd.DataFrame()
+        df["type"] = [ev.__class__.__name__ for dim in spec_dims for ev in dim.events]
+        df["label"] = [ev.label for dim in spec_dims for ev in dim.events]
+        df["duration"] = [
+            ev.duration if ev.__class__.__name__ == "ConstantDurationEvent" else np.nan
+            for dim in spec_dims
+            for ev in dim.events
+        ]
+        df["fraction"] = [
+            ev.fraction if ev.__class__.__name__ == "SpectralEvent" else np.nan
+            for dim in spec_dims
+            for ev in dim.events
+        ]
```

Comment:
These are similar codes. You could simplify and compact this. Create a list of attributes ``attributes = ["duration", "fraction"]`` and a list of corresponding class name to looks for, cls_valid = ["ConstantDurationEvent", "SpecctralEvent"]``, then use a for loop.

```
for attr, cls_name in zip(attributes, cls_valid):
    df[attr] = [
        getattr(ev, attr) if ev.__class__.__name__ == cls_name else np.nan
        for dim in spec_dims
        for ev in dim.events
    ]
```